legacy-sdk/whirlpool/src/impl/whirlpool-impl.ts

```diff
@@ -599,14 +598,18 @@ export class WhirlpoolImpl implements Whirlpool {
     const shouldDecreaseLiquidity = positionData.liquidity.gtn(0);
 
     const rewardsToCollect = this.data.rewardInfos
-      .filter((_, i) => {
+      .map((info, i) => ({ info, index: i }))
+      .filter(({ info, index }) => {
+        if (info.mint.equals(PublicKey.default)) {
```

Comment:
we can use PoolUtil.isRewardInitialized here

Reply:
Good point, fixed!